File: Python/Trabalho_POO_parte_2/MAIN.py
from Simulador import *
import time

menu = Simulador(20)
opcao = 0
dado_int_1 = 0 
dado_int_2 = 0
dado_float = 0.0
dado_string = ""
# OBS: METODOS COMO final, INTERFACE, instanceof. não são possivel de serem feitos manualmente, pois variasi ferramentas não existe ou ja são feitas implicitamente por propria interpretação da linguagem
# self = this
# qualquer variavel criado fora de um construtor em um objeto  é uma variavel static
while opcao != 15:
    print("\nMENU - Escolha uma das opcoes abaixo")
    print("1  - Incluir veiculo na pista")
    print("2  - Remover veiculo na pista")
    print("3  - Abastecer veiculo")
    print("4  - Movimentar veiculo")
    print("5  - Mo4vimentar veiculos por tipo")
    print("6  - Movimentar todos os veiculos")
    print("7  - Imprimir todos os dados de todos os veiculos")
    print("8  - Imprimir todos os dados de veiculos por tipo")
    print("9  - Calibrar um pneu especifico")
    print("10 - Calibrar todos os pneus de um veiculo especifico")
    print("11 - Esvaziar todos os pneus de um veiculo especifico")
    print("12 - Imprimir pista de corrida")
    print("13 - Gravar veiculos em arquivo")
    print("14 - Ler veiculos do arquivo")
    print("15 - Sair da aplicacao\n")

    try:
        opcao = int(input("Digite sua opcao: "))

        if opcao == 1:
            print("\tB - Bicicleta")
            print("\tM - Motocicleta")
            print("\tC - Carro Passeio")
            print("\tE - Carro esportivo")
            dado_string = str(input("\tQue tipo de veiculo você quer incluir?"))
            if dado_string in ["E", "B", "M", "C"]:
                try:
                    menu.incluir_veiculo(dado_string)
                except ValueError:
                    print("Opção não existente")
            else:
                print("Opção inválida")

        elif opcao == 2:
            dado_int_1 = int(input("Em qual veiculo você quer remover? Informe o ID: "))
            menu.remover_veiculo(dado_int_1)

        elif opcao == 3:
            dado_int_1 = int(input("Em qual veiculo você quer abastecer? Informe o ID: "))
            menu.abastecer_veiculo(dado_int_1)

        elif opcao == 4:
            dado_int_1 = int(input("Em qual veiculo você quer mover? Informe o ID: "))
            menu.mover_veiculo(dado_int_1)

        elif opcao == 5:
            print("\tB - Bicicleta")
            print("\tM - Motocicleta")
            print("\tC - Carro Passeio")
            print("\tE - Carro esportivo")
            dado_string = str(input("\tQue tipo de veiculo vc quer mover?"))
            menu.mover_veiculos_tipo(dado_string)

        elif opcao == 6:
            menu.mover_todos_veiculos()

        elif opcao == 7:
            menu.mostrar_dados_todos_veiculos()

        elif opcao == 8:
            dado_string = str(input("Mostrar dados dos veiculos de qual tipo: "))
            menu.mostrar_dados_veiculos_tipo(dado_string)

        elif opcao == 9:
            dado_int_1 = int(input("Qual veiculo você quer mexer nos pneus? Informe o ID: "))
            menu.veiculo_mexer_pneu(dado_int_1)

        elif opcao == 10:
            dado_string = str(input("Mostrar dados dos veiculos de qual tipo: "))
            menu.calibrar_veiculos_tipo(dado_string)

        elif opcao == 12:
            menu.imprimir_pista()

        elif opcao == 13:
            menu.gravar_arquivo()

        elif opcao == 14:
            menu.ler_arquivo()

    except ValueError:
        print("Opcao não existente tente novamente")

    time.sleep(3)

print("Encerrando a aplicacao.")

File: C++/Trabalho_POO_parte_2/MAIN.cpp
#include <iostream>
#include <string>
#include <chrono>
#include <thread>
#include "Simulador.cpp"

int main()
{
    Simulador menu(20);
    int opcao = 0;
    int dado_int_1 = 0;
    int dado_int_2 = 0;
    float dado_float = 0.0;
    char dado_char = 'a';

    while (opcao != 15)
    {
        std::cout << "\nMENU - Escolha uma das opcoes abaixo" << std::endl;
        std::cout << "1  - Incluir veiculo na pista" << std::endl;
        std::cout << "2  - Remover veiculo na pista" << std::endl;
        std::cout << "3  - Abastecer veiculo" << std::endl;
        std::cout << "4  - Movimentar veiculo" << std::endl;
        std::cout << "5  - Movimentar veiculos por tipo" << std::endl;
        std::cout << "6  - Movimentar todos os veiculos" << std::endl;
        std::cout << "7  - Imprimir todos os dados de todos os veiculos" << std::endl;
        std::cout << "8  - Imprimir todos os dados de veiculos por tipo" << std::endl;
        std::cout << "9  - Calibrar um pneu especifico" << std::endl;
        std::cout << "10 - Calibrar todos os pneus de um veiculo especifico" << std::endl;
        std::cout << "11 - Esvaziar todos os pneus de um veiculo especifico" << std::endl;
        std::cout << "12 - Imprimir pista de corrida" << std::endl;
        std::cout << "13 - Gravar veiculos em arquivo" << std::endl;
        std::cout << "14 - Ler veiculos do arquivo" << std::endl;
        std::cout << "15 - Sair da aplicacao" << std::endl;
        std::cout << std::endl;

        try
        {
            std::cout << "Digite sua opcao: ";
            std::cin >> opcao;

            if (opcao == 1)
            {
                std::cout << "\tB - Bicicleta" << std::endl;
                std::cout << "\tM - Motocicleta" << std::endl;
                std::cout << "\tC - Carro Passeio" << std::endl;
                std::cout << "\tE - Carro esportivo" << std::endl;
                std::cout << "Que tipo de veiculo você quer incluir? ";
                std::cin >> dado_char;

                if (dado_char == 'E' || dado_char == 'B' || dado_char == 'M' || dado_char == 'C')
                {
                    try
                    {
                        menu.incluir_veiculo(dado_char);
                    }
                    catch (const std::invalid_argument &e)
                    {
                        std::cout << "Opção não existente" << std::endl;
                    }
                }
                else
                {
                    std::cout << "Opção inválida" << std::endl;
                }
            }
            else if (opcao == 2)
            {
                std::cout << "Em qual veiculo você quer remover? Informe o ID: ";
                std::cin >> dado_int_1;
                menu.remover_veiculo(dado_int_1);
            }
            else if (opcao == 3)
            {
                std::cout << "Em qual veiculo você quer abastecer? Informe o ID: ";
                std::cin >> dado_int_1;
                menu.abastecer_veiculo(dado_int_1);
            }
            else if (opcao == 4)
            {
                std::cout << "Em qual veiculo você quer mover? Informe o ID: ";
                std::cin >> dado_int_1;
                menu.mover_veiculo(dado_int_1);
            }
            else if (opcao == 5)
            {
                std::cout << "\tB - Bicicleta" << std::endl;
                std::cout << "\tM - Motocicleta" << std::endl;
                std::cout << "\tC - Carro Passeio" << std::endl;
                std::cout << "\tE - Carro esportivo" << std::endl;
                std::cout << "Que tipo de veiculo você quer mover? ";
                std::cin >> dado_char;
                menu.mover_veiculos_tipo(dado_char);
            }
            else if (opcao == 6)
            {
                menu.mover_todos_veiculos();
            }
            else if (opcao == 7)
            {
                menu.mostrar_dados_todos_veiculos();
            }
            else if (opcao == 8)
            {
                std::cout << "Mostrar dados dos veiculos de qual tipo: ";
                std::cin >> dado_char;
                menu.mostrar_dados_veiculos_tipo(dado_char);
            }
            else if (opcao == 9)
            {
                std::cout << "Qual veiculo você quer mexer nos pneus? Informe o ID: ";
                std::cin >> dado_int_1;
                menu.veiculo_mexer_pneu(dado_int_1);
            }
            else if (opcao == 10)
            {
                std::cout << "Mostrar dados dos veiculos de qual tipo: ";
                std::cin >> dado_char;
                menu.calibrar_veiculos_tipo(dado_char);
            }
            else if (opcao == 12)
                menu.imprimir_pista();
            else if (opcao == 13)
                menu.gravar_arquivo();
            else if (opcao == 14)
                menu.ler_arquivo();
        }
        catch (const std::invalid_argument &e)
        {
            std::cout << "Opcao não existente, tente novamente" << std::endl;
        }
    }
    std::cout << "Encerrando a aplicacao." << std::endl;

    return 0;
}
